### ISSUE TITLE
Archived asset cannot be favorite

### ISSUE THREAD

**Issue by samip5** [immich-app/immich#2287]
### The bug

I will deem this a bug as archived asset gets favorite tag removed and it cannot be set.

![Screenshot_20230419-092044](https://user-images.githubusercontent.com/1703002/232984893-fd3dd20d-455f-4705-a118-913732b75d89.png)

### The OS that Immich Server is running on

Kubernetes

### Version of Immich Server

1.54.0

### Version of Immich Mobile App

1.54.0

### Platform with the issue

- [X] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Not relevant
```

### Your .env content

```Shell
Not relevant
```

### Reproduction steps

```bash
1. Find an asset and set it as favorite
2. Check that it can be found in favorites view
3. Archive it
4. Notice that you cannot find it in favorites anymore and asset debug info show that it's not favourite anymore.
```

### Additional information

The screenshot is from when I starred it without leaving the viewer but it's not saved to the asset.

**Comment by alextran1502** [comment]
Confirm that this is a bug on mobile

**Comment by truedem0n** [comment]
I would like to work on this one.

**Comment by fyfrey** [comment]
I think this is fixed with #2413. Favorites that are archived are visible in both the favorite view and the archive view
